aws-athena/views/default-vw_pin_sale.sql

```diff
@@ -60,7 +60,8 @@ unique_sales AS (
             NULLIF(sales.instrtyp, '') AS deed_type,
             -- "nopar" is number of parcels sold
             NOT COALESCE(
-                sales.nopar <= 1 AND calculated.nopar_calculated = 1,
+                (sales.nopar <= 1 OR sales.nopar IS NULL)
+                AND calculated.nopar_calculated = 1,
                 FALSE
```

Comment:
**suggestion:** The double negative in the logic here is confusing. You might want to refactor it.

Reply:
Refactored. Counts still look good:

```
with correct as (
select
    'now' as new,
    (num_parcels_sale > 1 and is_multisale) OR (num_parcels_sale = 1 and NOT is_multisale) as "multi_sale matches num_parcels_sale"
from "z_ci_335-address-null-values-for-iasworldsalesnopar-in-construction-of-defaultvw-pin-saleis-multisale_default".vw_pin_sale

UNION ALL

select
    'before' as new,
    (num_parcels_sale > 1 and is_multisale) OR (num_parcels_sale = 1 and NOT is_multisale) as correct
from default.vw_pin_sale
)

select new, "multi_sale matches num_parcels_sale", count(*) as count from correct
group by "multi_sale matches num_parcels_sale", new
order by new
```

new | multi_sale matches num_parcels_sale | count
-- | -- | --
before | FALSE | 4
before | TRUE | 2496069
now | TRUE | 2496073